pkg/operator/ceph/cluster/mgr/mgr.go

```diff
@@ -160,9 +160,12 @@ func (c *Cluster) Start() error {
 				return errors.Wrapf(err, "failed to create mgr deployment %s", resourceName)
 			}
 			logger.Infof("deployment for mgr %s already exists. updating if needed", resourceName)
-
-			if err := updateDeploymentAndWait(c.context, c.clusterInfo, d, config.MgrType, mgrConfig.DaemonID, c.spec.SkipUpgradeChecks, false); err != nil {
-				logger.Errorf("failed to update mgr deployment %q. %v", resourceName, err)
+			if err := updateDeploymentAndWait(c.context, c.clusterInfo, d, config.MgrType, mgrConfig.DaemonID, c.spec.SkipUpgradeChecks, c.spec.ContinueUpgradeAfterChecksEvenIfNotHealthy); err != nil {
```

Comment:
Testing this change worked as expected for me. 

Something that helped test it locally was to change shorten the retry `attempts` from 100 --> 10 so it would timeout waiting for the pending pod only 30s instead of 300s, and also add this log line at the start of [WaitForDeploymentToStart()](https://github.com/rook/rook/blob/master/pkg/operator/k8sutil/deployment.go#L118-L121)
```go
	attempts := 10
	logger.Infof("waiting for deployment %q to start", deployment.Name)
```

Reply:
Sounds good to remove the test changes!

Reply:
> Sounds good to remove the test changes!

done, thanks!